crates/turbopack-css/src/chunk/writer.rs

```diff
@@ -1,108 +1,70 @@
-use std::{collections::VecDeque, fmt::Write};
+use std::{collections::VecDeque, io::Write};
 
+use anyhow::Result;
 use turbo_tasks::{primitives::StringVc, ValueToString};
+use turbopack_core::code_builder::{CodeBuilder, CodeVc};
 
-use super::CssImport;
-use crate::CssChunkItemContentVc;
+use super::{CssChunkItemVc, CssImport};
 
-pub async fn expand_imports<T: Write>(
-    writer: &mut WriterWithIndent<T>,
-    content_vc: CssChunkItemContentVc,
-) -> anyhow::Result<Vec<StringVc>> {
-    let content = &*content_vc.await?;
+#[turbo_tasks::value]
+#[derive(Default)]
+pub struct ExpandImportsResult {
+    pub external_imports: Vec<StringVc>,
+    pub codes: Vec<CodeVc>,
+}
+
+#[turbo_tasks::function]
+pub async fn expand_imports(chunk_item: CssChunkItemVc) -> Result<ExpandImportsResultVc> {
+    let content = chunk_item.content().await?;
     let mut stack = vec![(
-        content_vc,
+        chunk_item,
         content.imports.iter().cloned().collect::<VecDeque<_>>(),
-        (0, "".to_string()),
+        "".to_string(),
     )];
-    let mut external_imports = vec![];
+    let mut result = ExpandImportsResult {
+        external_imports: vec![],
+        codes: vec![],
+    };
 
-    while let Some((content_vc, imports, (indent, close))) = stack.last_mut() {
+    while let Some((chunk_item, imports, close)) = stack.last_mut() {
         match imports.pop_front() {
             Some(CssImport::Internal(import, imported_chunk_item)) => {
-                let (open, inner_indent, close) = import.await?.attributes.await?.print_block()?;
+                let (open, close) = import.await?.attributes.await?.print_block()?;
 
                 let id = &*imported_chunk_item.to_string().await?;
+                let mut code = CodeBuilder::default();
```

Comment:
I think we can extract this so we only use a single `CodeBuilder`? Instead of storing `codes` vec, we store just a `code`.